src/background.js

```diff
@@ -28,6 +33,32 @@ browser.browserAction.onClicked.addListener(() => {
     openOverview()
 })
 
+
+browser.bookmarks.onCreated.addListener((id, bookmarkInfo) => {
+    //TO-DO Add method to handle indexing of bookmark
+    //bookmarkInfo.url gives the URL of the bookmark
+    //id is the unique id of the bookmark
+    //https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/bookmarks/onCreated
+    //Check if there already exists a page with this URL.
+    const samePageCandidates = (await findPagesByUrl({url})).rows.map(row => row.doc);
+    const reusablePage = await tryReidentifyPage({null, bookmarkInfo.url, samePageCandidates})
+    const pageData = fetchPageData(bookmarkInfo.url)
+    const favIconAttachment = formatFavIconAttachment(pageData.favIconURI)
+    const freezeDryBlob = new Blob([pageData.freezeDryHTML], {type: 'text/html;charset=UTF-8'})
+    const pageDoc = {
+        url: bookmarkInfo.url,
+        content: pageData.content,
+        _id: getPageId(pageData),
+        _attachments: {
```

Comment:
Looks good here with the doc structure and fields. With `_attachments`, Pouch will expect each one to be an object of at least shape: `{ content_type: string, data: Blob }`. For `content_type`, if you already have a `Blob` instance, you can use `.type` property to get the `content_type`. So here `content_type: freezeDryBlob.type`.

Also instead of importing `formatFavIconAttachment`, that function just wraps around a call to `dataURLToBlob` from `blob-util` package. So I think better to use that directly here on the `pageData.favIconURI`.